### DISCUSSION TITLE
Frigate begins recording clip of moving object then begins tracking stationary leading to long clips

### DISCUSSION THREAD

**Discussion by biggen1684** [blakeblackshear/frigate#460] [Ask A Question] [answered]
I'm monitoring a roadway and having Frigate save clips for only the `object` of `cars`.  I setup a mask to have it block out anything that isn't on the roadway.  What I'm finding is that Frigate correctly IDs and tracks cars moving on the road and writes clips for that.  But then sometimes it will lock onto the white truck that is parked across the street and continue writing to the same clip for long amounts of time.  So I'm getting clips that are several minutes long even though the originally tracked car is long gone.  I'm confused on why Frigate even sees that white truck since it is behind the mask.  Here are a few pictures to help explain.  First is the overall scene:

![latest (1)](https://user-images.githubusercontent.com/26392652/103424543-8f738680-4b72-11eb-8e93-c83c68297b35.jpg)

I setup a mask just to expose the road.  Here is the mask:

![Screenshot from 2020-12-31 14-15-58](https://user-images.githubusercontent.com/26392652/103424600-d792a900-4b72-11eb-851b-4af68a654325.png)

Here are a few clips from HA to show what I'm talking about.  You can see a few of the clips have the parked white truck across the street inside bounding boxes and the clip lengths are very long.  The originally tracked car moved through and out of the scene within a handful of seconds yet Frigate continues to keep the clip open.  

I know I can force Frigate to a certain maximum length of clips but that doesn't sound like a great resolution.  Shouldn't it be ignoring any non-moving `car` that is also behind a mask??  Is there a better way to do what I want to do if not??

![Screenshot from 2020-12-31 14-19-39](https://user-images.githubusercontent.com/26392652/103424752-d6ae4700-4b73-11eb-967e-3aa04c83832e.png)

**Comment by blakeblackshear** [comment, marked as answer]
This is expected behavior at the moment. The mask has nothing to do with object detection. It only applies to motion detection. When a tracked object enters the masked area, frigate will continue tracking it and pick up anything in the background as well. Object detection masks will be a separate feature.

**Comment by biggen1684** [comment]
Ok, thanks Blake.  I'll go ahead and set the `max_seconds: 300` to something lower.
